packages/flutter/test/widgets/routes_test.dart

```diff
@@ -1873,6 +1874,272 @@ void main() {
       await tester.pump();
       expect(tester.takeException(), null);
     });
+
+    testWidgets('outgoing route receives a delegated transition from the new route', (WidgetTester tester) async {
+      final GlobalKey<NavigatorState> navigatorKey = GlobalKey<NavigatorState>();
+
+      final MaterialPageRoute<void> materialPageRoute = MaterialPageRoute<void>(
+        builder: (BuildContext context) {
+          return Scaffold(
+            body: TextButton(
+              onPressed: () {
+                final CupertinoPageRoute<void> route = CupertinoPageRoute<void>(
+                  builder: (BuildContext context) {
+                    return  const Text('Cupertino Transition');
+                  }
+                );
+                Navigator.of(context).push(route);
+              },
+              child: const Text('Cupertino Transition'),
+            ),
+          );
+        }
+      );
+
+      await tester.pumpWidget(MaterialApp(
+        navigatorKey: navigatorKey,
+        theme: ThemeData(
+          pageTransitionsTheme: const PageTransitionsTheme(
+            builders: <TargetPlatform, PageTransitionsBuilder>{
+              TargetPlatform.iOS: ZoomPageTransitionsBuilder(),
+              TargetPlatform.android: ZoomPageTransitionsBuilder(),
+            },
+          ),
+        ),
+        home: Scaffold(
+          body: TextButton(
+            onPressed: () {
+              navigatorKey.currentState!.push<void>(materialPageRoute);
+            },
+            child: const Text('Material Route Transition'),
+          ),
+        ),
+        )
+      );
+
+      expect(materialPageRoute.receivedTransition, null);
+
+      await tester.tap(find.text('Material Route Transition'));
+
+      await tester.pumpAndSettle();
+
+      expect(find.text('Cupertino Transition'), findsOneWidget);
+      expect(find.text('Material Route Transition'), findsNothing);
+
+      expect(materialPageRoute.receivedTransition, null);
+
+      await tester.tap(find.text('Cupertino Transition'));
+
+      await tester.pumpAndSettle();
+
+      expect(materialPageRoute.receivedTransition, isNotNull);
+      expect(materialPageRoute.receivedTransition, CupertinoPageTransition.delegateTransition);
+    });
+
+    testWidgets('outgoing route does not receive a delegated transition from a route with the same transition', (WidgetTester tester) async {
```

Comment:
This just makes me wonder: it's not that the routes are the same type, it's that the routes' delegatedTransition is the same as its usual outgoing transition? Or how does that work?

Reply:
This makes sense now with the DelegatedTransition subclassing.